django/thunderstore/repository/admin.py

```diff
@@ -78,6 +79,24 @@ def get_readonly_fields(self, request, obj=None):
     search_fields = ("name",)
 
 
+@admin.register(Namespace)
+class NamespaceAdmin(admin.ModelAdmin):
+    def get_readonly_fields(self, request, obj=None):
+        if obj:
+            return self.readonly_fields
+        else:
+            return []
+
+    actions = (
+        activate,
+        deactivate,
+    )
+    readonly_fields = ("name",)
+    list_display = ("name", "is_active", "team")
+    list_filter = ("is_active",)
+    search_fields = ("name", "team__name")
```

Comment:
Given that the entire model is supposed to be read only, add
```python
    def has_add_permission(self, request: HttpRequest, obj) -> bool:
        return False

    def has_edit_permission(self, request: HttpRequest, obj=None) -> bool:
        return False

    def has_delete_permission(self, request: HttpRequest, obj=None) -> bool:
        return False
```

Reply:
Doned